app/src/test/java/org/simple/clinic/settings/SettingsEffectHandlerTest.kt

```diff
@@ -40,7 +46,7 @@ class SettingsEffectHandlerTest {
         phone = "1234567890"
     )
     val updatedUser = savedUser.copy(fullName = "Mahalakshmi T Sharma", phoneNumber = "0987654321")
```

Comment:
Can we not expose the copy constructor and use a reducer function instead?
`savedUser.copy(fullName = "Mahalakshmi T Sharma", phoneNumber = "0987654321")`

Reply:
I didn't want to add this because that function will only be used in tests. However, I could make it an extension function that's private to this test. I'm not sure I'd want to do this since it would only be used in one place, however.

Reply:
Never mind, I'll add the reducer functions since there are places we can use it in production. We can change the production usages later.

Reply:
Copy constructors will cause flaky tests in case the implementation details change later. We should write a lint rule that flags any copy constructor invocations outside the data class.

Reply:
Done.